monitoring/nodesstatus.go

```diff
@@ -89,71 +91,127 @@ func (r *nodesStatusChecker) Check(ctx context.Context, reporter health.Reporter
 	}
 }
 
+// NodeStatusCheckerConfig is the Kubernetes node status checker configuration.
+type NodeStatusCheckerConfig struct {
+	// KubeConfig provides Kubernetes access.
+	KubeConfig
+	// NodeName is the Kubernetes node name.
+	NodeName string
+	// Conditions is a list of Kubernetes node conditions to monitor.
```

Comment:
This still lacks context information about the type of conditions this can handle - namely, conditions whose failure is triggered by having the `status` evaluate to `v1.ConditionTrue` and not arbitrary conditions.

Also, this is somewhat limiting as a result. I can see this being useful for handling arbitrary conditions - what would it take to generalize for that? I think that something like `IsConditionTriggered(v1.NodeConditionType) bool` could work.
You would provide a custom implementation as a fall-back:
```go
func IsNPDConditionTriggered(cond v1.NodeConditionType) bool {
	return r.isNotReadyCondition(cond) || r.isFailureCondition(cond)
}
```
then any other condition could be composable as:
```go
func customIsConditionTriggered(cond v1.NodeConditionType) bool {
	return monitoring.IsNDPConditionTriggered(cond) || isAnotherConditionTriggered(cond)
}
```